### ISSUE TITLE
Docker cp command is not working on Ubuntu 16.04 in Azure Marketplace

### ISSUE THREAD

**Issue by jbmcfarlin31** [moby/moby#40211]
**Description**
Running Docker `v19.03.4` on Ubuntu 16.04 images within the Azure marketplace and Azure stack marketplace result in the error:
```
failed to augment data: API error (500): error processing tar file: docker-tar: relocation error: /lib/x86_64-linux-gnu/libnss_files.so.2: symbol __libc_readline_unlocked, version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
: exit status 127
```

When trying to use `docker cp`, the error above occurs only on Azure marketplace image versions of Ubuntu 16.04. 

**Steps to reproduce the issue:**
1. `docker run --rm -d --name nginx nginx:1.16`
2. `docker cp nginx:/etc/hostname .`

**Describe the results you received:**
The following error is returned:
```
Error response from daemon: error processing tar file: docker-tar: relocation error: /lib/x86_64-linux-gnu/libnss_files.so.2: symbol __libc_readline_unlocked, version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
: exit status 127
```

**Describe the results you expected:**
I expected a successful copy from container to host.

**Additional information you deem important (e.g. issue happens only occasionally):**
This appears to be a regression of #39449 

**Output of `docker version`:**
```
Client: Docker Engine - Community
 Version:           19.03.4
 API version:       1.40
 Go version:        go1.12.10
 Git commit:        9013bf583a
 Built:             Fri Oct 18 15:53:51 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.4
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.10
  Git commit:       9013bf583a
  Built:            Fri Oct 18 15:52:23 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
 Containers: 49
  Running: 41
  Paused: 0
  Stopped: 8
 Images: 80
 Server Version: 19.03.4
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.15.0-1013-azure
 Operating System: Ubuntu 16.04.4 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 13.69GiB
 Name: <redacted>
 ID: ZH4B:XEYB:HHVZ:LOXO:U2HA:VIZC:6KGP:LAAP:MZ4H:5GAI:46ZS:UCVS
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
The output above was executed on an Ubuntu 16.04 image from the Azure Stack marketplace, however, this also occurs on the same Ubuntu 16.04 image within Public Azure marketplace.

**Comment by thaJeztah** [comment]
Thanks for reporting; is this on a fresh install of docker, or a docker install that was upgraded from an older version?

**Comment by thaJeztah** [comment]
ping @tonistiigi @justincormack @cpuguy83 PTAL

**Comment by jbmcfarlin31** [comment]
> Thanks for reporting; is this on a fresh install of docker, or a docker install that was upgraded from an older version?

@thaJeztah this was on a brand new install of Docker `19.03.4` - no upgrade.

**Comment by amirschw** [comment]
Happens to me with Docker version 19.03.5 on Ubuntu 16.04.6 LTS:

```
$ docker cp nginx:/etc/hostname .
Error response from daemon: error processing tar file: docker-tar: relocation error: /lib/x86_64-linux-gnu/libnss_files.so.2: symbol __libc_readline_unlocked, version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference
: exit status 127
/usr/bin/docker failed with return code: 1
```

**Comment by jbmcfarlin31** [comment]
@thaJeztah has there been any fix towards this? Or an update? Thanks!

**Comment by thaJeztah** [comment]
Still being looked at. There's a few options, but some of those have side-effects, which is currently being discussed

**Comment by thaJeztah** [comment]
Apologies for the delay; we just shipped Docker 19.03.8 (https://github.com/docker/docker-ce/releases/tag/v19.03.8), which has a fix for this issue in this commit: https://github.com/moby/moby/commit/aa6a9891b09cce3d9004121294301a30d45d998d, which is a follow up to https://github.com/moby/moby/pull/39612, with additional changes to address CVE-2019-14271 on certain configurations. Packages are up on download.docker.com, and the [official images on Docker Hub](https://hub.docker.com/_/docker) should be available soon.

Thanks again for reporting; I'll close this issue, but feel free to continue the  after I closed

**Comment by jbmcfarlin31** [comment]
> Apologies for the delay; we just shipped Docker 19.03.8 (https://github.com/docker/docker-ce/releases/tag/v19.03.8), which has a fix for this issue in this commit: [aa6a989](https://github.com/moby/moby/commit/aa6a9891b09cce3d9004121294301a30d45d998d), which is a follow up to #39612, with additional changes to address CVE-2019-14271 on certain configurations. Packages are up on download.docker.com, and the [official images on Docker Hub](https://hub.docker.com/_/docker) should be available soon.
> 
> Thanks again for reporting; I'll close this issue, but feel free to continue the after I closed

@thaJeztah Just wanted to relay that I verified on Azure's Ubuntu 16.04 image - docker version 19.03.8 worked great. Thanks for the update!

**Comment by thaJeztah** [comment]
Perfect! Thanks for confirming!
